### PR TITLE
feat: [CDS-54457]: Adding InstanceSyncPerpetualTaskMapping collection for NG intance sync V2

### PR DESCRIPTION
**Pull request by namantalaycha10**
…for NG intance sync V2

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45224)

### CODE REVIEW

File: 126-instance/src/main/java/io/harness/entities/InstanceSyncPerpetualTaskMapping.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+                 .unique(true)
+                 .name("unique_connectorIdentifier_perpetualTaskId_idx")
+                 .field(InstanceSyncPerpetualTaskMappingKey.connectorIdentifier)
+                 .field(InstanceSyncPerpetualTaskMappingKey.perpetualTaskId)
+                 .build())
+        .build();
+  }
+
+  @Id @dev.morphia.annotations.Id String id;
+  @FdUniqueIndex String connectorIdentifier;
```
Comment (abosii-harness) [inline 126-instance/src/main/java/io/harness/entities/InstanceSyncPerpetualTaskMapping.java:49]: Do we have to use this FdUniqueIndex if you're creating mongo indexes manually?

File: 126-instance/src/main/java/io/harness/serializer/morphia/InstanceMorphiaClassesRegistrar.java [hunk 2]
```diff
@@ -10,10 +10,7 @@
 import static io.harness.annotations.dev.HarnessTeam.DX;
 
 import io.harness.annotations.dev.OwnedBy;
-import io.harness.entities.DeploymentAccounts;
-import io.harness.entities.DeploymentSummary;
-import io.harness.entities.InfrastructureMapping;
-import io.harness.entities.Instance;
+import io.harness.entities.*;
```
Comment (abosii-harness) [inline 126-instance/src/main/java/io/harness/serializer/morphia/InstanceMorphiaClassesRegistrar.java:13]: We should extend this import

File: 126-instance/src/main/java/io/harness/entities/InstanceSyncPerpetualTaskMapping.java (hunk 1 above)
Comment (namantalaycha10) [reply 126-instance/src/main/java/io/harness/entities/InstanceSyncPerpetualTaskMapping.java]: Done

File: 126-instance/src/main/java/io/harness/serializer/morphia/InstanceMorphiaClassesRegistrar.java (hunk 2 above)
Comment (namantalaycha10) [reply 126-instance/src/main/java/io/harness/serializer/morphia/InstanceMorphiaClassesRegistrar.java]: Done

**Pull request closed without merging**